src/components/MenuPanel/CombinedPanel.js

```diff
@@ -0,0 +1,274 @@
+                            autoClosing
+                              ? lerp(
+                                  v,
+                                  -(
+                                    ACTIVITY_PANEL_WIDTH +
+                                    ACTIVITY_PANEL_SHADOW_WIDTH
+                                  ),
+                                  0
+                                )
+                              : lerp(
+                                  v,
+                                  -(
+                                    ACTIVITY_PANEL_WIDTH -
+                                    MENU_PANEL_WIDTH +
+                                    ACTIVITY_PANEL_SHADOW_WIDTH
+                                  ),
+                                  MENU_PANEL_WIDTH
+                                )
+                          }px, 0, 0)`
+                      ),
+                    }}
+                  >
+                    <ActivityPanel
+                      apps={this.props.apps}
+                      displayBackButton={autoClosing}
+                      onClearAll={onClearActivities}
+                      onClose={this.handleActivityClose}
+                      open={activityOpened}
+                      shouldClose={this.shouldCloseActivityPanel}
+                    />
+                  </animated.div>
+                  {autoClosing && (
+                    <Overlay
+                      onClick={onMenuPanelClose}
+                      style={{
+                        zIndex: '2',
+                        /* by leaving a 1px edge Android users can swipe to open
+                         * from the edge of their screen when an iframe app is being
+                         * used */
+                        width: combinedPanelProgress.interpolate(p =>
+                          p === 0 ? '1px' : '100vw'
+                        ),
+                        opacity: combinedPanelProgress,
+                      }}
+                    />
+                  )}
+                  {children}
+                </React.Fragment>
+              )
+            }}
+          </Spring>
+        )}
+      </SwipeContainer>
+    )
+  }
+}
+
+const Overlay = styled(animated.div)`
+  position: absolute;
+  height: 100vh;
+  background: rgba(0, 0, 0, 0.3);
+`
+
+const Wrap = styled(animated.div)`
+  width: ${MENU_PANEL_WIDTH}px;
+  height: 100vh;
+  flex: none;
+`
+
+export default props => {
+  const {
+    unreadActivityCount,
+    clearActivities,
+    markActivitiesRead,
+  } = useContext(ActivityContext)
+
+  const requestClearActivities = useCallback(() => {
+    clearActivities()
+  }, [clearActivities])
+
+  const requestMarkActivitiesRead = useCallback(() => {
+    markActivitiesRead()
+  }, [markActivitiesRead])
+
+  return (
+    <CombinedPanel
+      {...props}
+      unreadActivityCount={unreadActivityCount}
+      onClearActivities={requestClearActivities}
+      onMarkActivitiesRead={requestMarkActivitiesRead}
```

Comment:
Could we move all of these to be at the `ActivityPanel` level instead, since that's where they're needed?

We use `onMarkActivitiesRead()` here, but it seems like a good candidate to move to a `componentDidUpdate()` based on the `open` prop in `ActivityPanel`.